### DISCUSSION TITLE
Is there any way to directly access the IR node when having the corresponding annotated target?

### DISCUSSION THREAD

**Discussion by Wonicon** [chipsalliance/chisel#2008] [Q&A] [answered]
Hello, everyone. I am learning FIRRTL pass and Chisel annotation. I want to use annotation to locate the module I want to modify, but as the tutorial[1] shows, I will get a `XXXTarget` object when consuming annotations. This type of objects mainly contains String fields to indicate which FIRRTL node it is, but does not have a field to point to that node object. Besides brute-force searching, is there any way to elegantly access these node objects?

I have read the following tutorials:
[1] https://www.chisel-lang.org/chisel3/docs/explanations/annotations.html
[2] https://github.com/chipsalliance/firrtl/tree/master/src/main/scala/tutorial

**Comment by sequencer** [comment, marked as answer]
This feature was introduced in chipsalliance/firrtl#1603
I believe this is what you are looking for:
https://github.com/chipsalliance/firrtl/blob/master/src/main/scala/firrtl/analyses/IRLookup.scala
You can check the usage via test
https://github.com/chipsalliance/firrtl/blob/master/src/test/scala/firrtlTests/analyses/IRLookupSpec.scala

**Reply by Wonicon** [reply]
Thanks a lot
